### PR TITLE
Artifactory Release Lifecycle Management - Support release bundles deletion

### PR DESCRIPTION
**Pull request by RobiNino**
- [x] All [tests](https://github.com/jfrog/jfrog-client-go#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] All [static analysis checks](https://github.com/jfrog/jfrog-client-go/actions/workflows/analysis.yml) passed.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
-----

### CODE REVIEW

**Review by eyalbe4 — CHANGES_REQUESTED** [review]

**Review by yahavi — CHANGES_REQUESTED** [review]
Please add GetReleaseBundleVersionPromotions, DeleteReleaseBundleVersion and DeleteReleaseBundleVersionPromotion to the README.

Also, please add some unit tests.

File: lifecycle/services/delete.go [hunk 1]
```diff
@@ -51,5 +73,51 @@ func (rbs *ReleaseBundlesService) RemoteDeleteReleaseBundle(params distribution.
 		return err
 	}
 
-	return errorutils.CheckResponseStatusWithBody(resp, body, http.StatusAccepted)
+	err = errorutils.CheckResponseStatusWithBody(resp, body, http.StatusAccepted)
```
Comment (yahavi) [inline lifecycle/services/delete.go:76]: Let's add a debug logging with the server responses.

File: lifecycle/services/distribute.go [hunk 2]
```diff
@@ -102,3 +107,35 @@ type PathMapping struct {
...
+	}
+	httpClientsDetails := rbs.GetLifecycleDetails().CreateHttpClientDetails()
+	resp, body, _, err := rbs.client.SendGet(requestFullUrl, true, &httpClientsDetails)
+	if err != nil {
+		return
+	}
+	if err = errorutils.CheckResponseStatusWithBody(resp, body, http.StatusAccepted, http.StatusOK); err != nil {
+		return
+	}
+	err = errorutils.CheckError(json.Unmarshal(body, &distributionsResp))
```
Comment (yahavi) [inline lifecycle/services/distribute.go:125]: Let's add a debug logging with the server responses.

File: lifecycle/services/promote.go [hunk 3]
```diff
@@ -54,6 +57,77 @@ func (rbs *ReleaseBundlesService) Promote(rbDetails ReleaseBundleDetails, queryP
...
+	}
+	httpClientsDetails := rbs.GetLifecycleDetails().CreateHttpClientDetails()
+	resp, body, _, err := rbs.client.SendGet(requestFullUrl, true, &httpClientsDetails)
+	if err != nil {
+		return
+	}
+	if err = errorutils.CheckResponseStatusWithBody(resp, body, http.StatusOK); err != nil {
+		return
+	}
+	err = errorutils.CheckError(json.Unmarshal(body, &response))
```
Comment (yahavi) [inline lifecycle/services/promote.go:127]: Let's add a debug logging with the server responses.

**Review by yahavi — CHANGES_REQUESTED** [review]

**Pull request merged**